src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -470,6 +481,8 @@ private void Poll()
                     var consumed = _consumer.Consume(0);
                     if (consumed != null)
                         throw new IllegalActorStateException("Consumed message should be null");
+                    PausePartitions(_pausedPartitions);
+                    _pausedPartitions = ImmutableList<TopicPartition>.Empty;
```

Comment:
Fix: 

It is not possible to pause newly assigned partitions inside `OnPartitionsAssigned` event callback; `ConfluentKafka.ConsumerClient` will throw because the callback was invoked _before_ the internal partitions assignment list is updated. This is a hack to circumvent that, we call `Pause()` directly after the `Consume()` invocation.

Note that this is safe because everything is still running in the same thread.